### ISSUE TITLE
Unable to compile v0.2.11 with Rust 1.22.1

### ISSUE THREAD

**Issue by nahumclements** [crossbeam-rs/crossbeam#164]
I'm not able to compile crossbeam v0.2.11 using Rust 1.22.1. I get a compiler error:

```
    Updating registry `https://github.com/rust-lang/crates.io-index`
   Compiling crossbeam v0.2.11
error: expected one of `extern` or `fn`, found `auto`
  --> /opt/rust/.cargo/registry/src/github.com-1ecc6299db9ec823/crossbeam-0.2.11/src/mem/cache_padded.rs:54:12
   |
54 | pub unsafe auto trait ZerosValid {}
   |           -^^^^ unexpected token
   |           |
   |           expected one of `extern` or `fn` here

error: aborting due to previous error

error: Could not compile `crossbeam`.
```

Compiling with Rust 1.23 seems to work fine but I was surprised that a patch version change in crossbeam would mean I needed to change my Rust version.

**Comment by ignatenkobrain** [comment]
I tried to compile 0.3.1 and get same error.

**Comment by canndrew** [comment]
For the moment you can add `crossbeam = "=0.2.10" to your Cargo.toml files to fix this.

**Comment by stjepang** [comment]
Sorry about the fuss! I've published 0.2.12 and 0.3.2 with a fix. Let me know if the error persists.
